avm/res/cdn/profile/securityPolicies/main.bicep

```diff
@@ -0,0 +1,53 @@
+metadata name = 'CDN Profiles Security Policy'
+metadata description = 'This module deploys a CDN Profile Security Policy.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The resource name.')
+param name string
+
+@description('Conditional. The name of the parent CDN profile. Required if the template is used in a standalone deployment.')
+param profileName string
+
+@description('Required. Resource ID of WAF Policy.')
+param wafPolicyResourceId string
+
+// param associations associationsType
+@description('Required. Waf associations (see https://learn.microsoft.com/en-us/azure/templates/microsoft.cdn/profiles/securitypolicies?pivots=deployment-language-bicep#securitypolicywebapplicationfirewallassociation for details).')
+param associations associationsType
+
+resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
+  name: profileName
+}
+
+resource securityPolicies 'Microsoft.Cdn/profiles/securityPolicies@2024-02-01' = {
+  name: name
+  parent: profile
+  properties: {
+    parameters: {
+      type: 'WebApplicationFirewall'
+      wafPolicy: {
+        id: wafPolicyResourceId
+      }
+      associations: associations
+    }
+  }
+}
+
+type associationsType = {
```

Comment:
Hey @anderseide,
thank your for your contribution 🚀 

@gbeaud, I hope you don't mind me chiming in. Just one suggestion if I may:

IF you happen to add the `@export()` annotation to this type, you could actually create a UDT for the `securityPolicies` in the parent too, import the type there and just reuse it. This would look similar to
https://github.com/Azure/bicep-registry-modules/blob/e006349901e74388e1623b1572bc08a4f15edf75/avm/res/document-db/database-account/main.bicep#L914-L918

like
```bicep
@description('Optional. Array of Security Policy objects (see https://learn.microsoft.com/en-us/azure/templates/microsoft.cdn/profiles/securitypolicies for details).')
param securityPolicies securityPolicyType[]?

module profile_securityPolicies 'securityPolicies/main.bicep' = [
  for (securityPolicy, index) in (securityPolicies ?? []): {
    name: '${uniqueString(deployment().name)}-Profile-SecurityPolicy-${index}'
    params: {
      name: securityPolicy.name
      profileName: profile.name
      associations: securityPolicy.associations
      wafPolicyResourceId: securityPolicy.wafPolicyResourceId
    }
    dependsOn: [
      profile_afdEndpoints
      profile_customDomains
    ]
  }
]

import { associationsType } from 'securityPolicies/main.bicep'
type securityPolicyType = {
  @description('Required. (...)')
   name: string

  @description('Required. (...)')
   associations: associationsType 

  @description('Required. (...)')
   wafPolicyResourceId: string
}
```

Reply:
Good suggestion! I can update the code to use export if you like @gbeaud. I didn't use it because it's still marked as experimental, but if it has @AlexanderSehr's blessing I guess it's good enough 😃

Reply:
Hey @anderseide, could you elaborate which feature is experimental? Biceo type imports from other files should be 'good to go' by now (just not from e.g., the bicep-types repository). Or are you referring to the security policies? If the case, then I'd still argue for providing the UDT - but it would get us into a whole other conversation, so I'll hold onto that for now 😄

Reply:
I'm referring to the experimental flag `compileTimeImports` that must be enabled from the [Bicep config file](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-config#enable-experimental-features) as stated in the UDT [documentation here](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/user-defined-data-types#import-types-between-bicep-files). It's as far as I know it's required to be able to use `@export`. I see it's enabled for the AVM repo so maybe it's safe enough to use...

Reply:
Hey @anderseide,
I can't see it 😄 
https://github.com/Azure/bicep-registry-modules/blob/e006349901e74388e1623b1572bc08a4f15edf75/bicepconfig.json#L1-L12
Neither here, nor anywhere in the repository. I assume it was since moved out of the experimental state and the docs were not correctly updated. This would correlate to the mentioned Bicep version `0.21.X` as we're now at `0.29.4` if I'm not mistaken. As far as I'm concerend ... gogogo

Reply:
Uum, yeah, your are absolutely right! My bad 🤦 - I'll update the code, and I'll try to find time to send a PR for the docs as well...

Reply:
Updated code - running new deployment tests now. Also created PR for azure-docs